### PR TITLE
chore(deps): Bump `@metamask/signature-controller` from `^17.0.0` to `^19.1.0`, `@metamask/logging-controller` from `^3.0.0` to `^6.0.1`

### LINKED ISSUE #11170: SignatureController upgrade from v16 to v19
### What is this about?

The `SignatureController` is behind by three major versions and needs to be updated.
Upgrade from v16 to v19.

Because this controller is pretty far behind we expect it will have a number of breaking changes that will need to be dealt with so we haven't combined this with any other controller upgrades.

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### LINKED ISSUE #11378: LoggingController upgrade from v3 to v6
### What is this about?

The `LoggingController` is behind by three major versions and needs to be updated.
Upgrade from v3 to v6.

Because this controller is pretty far behind we expect it will have a number of breaking changes that will need to be dealt with so we haven't combined this with any other controller upgrades.

`LoggingController` is a peer dependency of `SignatureController`, and this ticket blocks `SignatureController` upgrades.

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by MajorLift**
### Changelog

```md
### Changed

- **BREAKING:** Bump `@metamask/signature-controller` from `^17.0.0` to `^19.1.0` ([#11169](https://github.com/MetaMask/metamask-mobile/pull/11169))
- **BREAKING:** Bump `@metamask/logging-controller` from `^3.0.0` to `^6.0.1` ([#11169](https://github.com/MetaMask/metamask-mobile/pull/11169))
```

### Diff analysis

- [@metamask/signature-controller@17.0.0...@metamask/signature-controller@19.0.0](https://github.com/MetaMask/core/compare/@metamask/signature-controller@17.0.0...@metamask/signature-controller@19.0.0#diff-f5256aab6986e16b893f20e4c0cb70424339ba4750ad31664c343f7dde11d064)
  - linter fixes
  - one change in [error handling]
- [@metamask/signature-controller@19.0.0...@metamask/signature-controller@19.1.0](https://github.com/MetaMask/core/compare/@metamask/signature-controller@19.0.0...@metamask/signature-controller@19.1.0#diff-f5256aab6986e16b893f20e4c0cb70424339ba4750ad31664c343f7dde11d064)
  - new non-required [`trace` constructor option](https://github.com/MetaMask/core/blob/main/packages/signature-controller/CHANGELOG.md#1910)
- [@metamask/logging-controller@3.0.0...@metamask/logging-controller@6.0.1](https://github.com/MetaMask/core/compare/@metamask/logging-controller@3.0.0...@metamask/logging-controller@6.0.1#diff-b144e88332de800bb1c113196a94d48417e4b183bcb2766250db3ceba69bc24e)
  - type updates for BaseControllerV2 alignment

## **Related issues**

- Closes https://github.com/MetaMask/metamask-mobile/issues/11378
- Closes https://github.com/MetaMask/metamask-mobile/issues/11170
- Same set of updates in extension https://github.com/MetaMask/metamask-extension/pull/27082

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -186,7 +186,7 @@
     "@metamask/scure-bip39": "^2.1.0",
     "@metamask/sdk-communication-layer": "0.29.0-wallet",
     "@metamask/selected-network-controller": "^15.0.2",
-    "@metamask/signature-controller": "^17.0.0",
+    "@metamask/signature-controller": "^19.1.0",
```
Comment (Cal-L) [inline package.json:189]: Looks like you mentioned there may be some breaking changes in the original issue - https://github.com/MetaMask/metamask-mobile/issues/11170. Was it confirmed that this was still functioning correctly without any other changes post bump?

**Review by Cal-L — COMMENTED** [review]
Left a comment!

File: package.json (hunk 1 above)
Comment (MajorLift) [reply package.json:189]: Good point! We added that as a general warning for updates spanning multiple major versions, but in this case the only breaking change that might cause issues is the `eth_sign` removal ([`signature-controller`](https://github.com/MetaMask/core/blob/main/packages/signature-controller/CHANGELOG.md#1700), [`logging-controller`](https://github.com/MetaMask/core/blob/main/packages/logging-controller/CHANGELOG.md#400)). Conveniently, the necessary updates for that has [already been handled in mobile](https://github.com/MetaMask/metamask-mobile/pull/11246). 

In extension, the same upgrade for these two controllers was merged into develop 10 days ago (https://github.com/MetaMask/metamask-extension/pull/27082/) and hasn't caused issues, so there aren't any obvious bugs with the controllers.

I did take a closer look at the diffs and it doesn't seem like there's cause for concern. I'll add this to the PR description.
- [@metamask/signature-controller@17.0.0...@metamask/signature-controller@19.1.0](https://github.com/MetaMask/core/compare/@metamask/signature-controller@17.0.0...@metamask/signature-controller@19.1.0#diff-f5256aab6986e16b893f20e4c0cb70424339ba4750ad31664c343f7dde11d064)
  - non-required feature additions (mostly related to the [`trace` constructor option](https://github.com/MetaMask/core/blob/main/packages/signature-controller/CHANGELOG.md#1910))
  - linter fixes
- [@metamask/logging-controller@3.0.0...@metamask/logging-controller@6.0.1](https://github.com/MetaMask/core/compare/@metamask/logging-controller@3.0.0...@metamask/logging-controller@6.0.1#diff-b144e88332de800bb1c113196a94d48417e4b183bcb2766250db3ceba69bc24e)
  - type updates for BaseControllerV2 alignment

Comment (MajorLift) [reply package.json:189]: Between v17 to v19, the only changes for `SignatureController` are linter fixes and one change in [error handling](https://github.com/MetaMask/core/compare/@metamask/signature-controller@17.0.0...@metamask/signature-controller@19.0.0#diff-f5256aab6986e16b893f20e4c0cb70424339ba4750ad31664c343f7dde11d064R434-R443).

- [@metamask/signature-controller@17.0.0...@metamask/signature-controller@19.0.0](https://github.com/MetaMask/core/compare/@metamask/signature-controller@17.0.0...@metamask/signature-controller@19.0.0#diff-f5256aab6986e16b893f20e4c0cb70424339ba4750ad31664c343f7dde11d064)

Comment (Cal-L) [reply package.json:189]: Thanks for confirming!

**Review by Cal-L — APPROVED** [review]
LGTM from platform!

**Review by sleepytanya — APPROVED** [review]

**Pull request merged**